### ISSUE TITLE
NoSuchMethodError when applying any rules

### ISSUE THREAD

**Issue by PatrickAuld** [nebula-plugins/gradle-resolution-rules-plugin#46]
I'm trying to apply the plugin to a multiproject Gradle build in a shared Gradle file that all Java projects import. Applying the plugin doesn't cause errors (but does print warning about no found rules) but the addition of the standard rules via `resolutionRules 'com.netflix.nebula:gradle-resolution-rules:latest.release'` creates the below method not found errors within Jackson. This seems like a chicken and the egg problem as I have to somehow figure out my dependencies within Gradle to get the dependency in my app together 😃 

I've played around with trying to set the Jackson version but haven't had much luck yet. If anyone know another fix for this it would help greatly. Thanks in advance.

> Failed to notify project evaluation listener.
>   > com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.getType()Lcom/fasterxml/jackson/databind/JavaType;
>   > com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.getType()Lcom/fasterxml/jackson/databind/JavaType;

... 12 more times.

> Caused by: java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.getType()Lcom/fasterxml/jackson/databind/JavaType;
>     at com.fasterxml.jackson.module.kotlin.KotlinNamesAnnotationIntrospector.hasCreatorAnnotation(KotlinModule.kt:81)
>     at com.fasterxml.jackson.databind.introspect.AnnotationIntrospectorPair.hasCreatorAnnotation(AnnotationIntrospectorPair.java:599)
>     at com.fasterxml.jackson.databind.deser.BasicDeserializerFactory._addDeserializerConstructors(BasicDeserializerFactory.java:401)
>     at com.fasterxml.jackson.databind.deser.BasicDeserializerFactory._constructDefaultValueInstantiator(BasicDeserializerFactory.java:325)
>     at com.fasterxml.jackson.databind.deser.BasicDeserializerFactory.findValueInstantiator(BasicDeserializerFactory.java:266)
>     at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.buildBeanDeserializer(BeanDeserializerFactory.java:266)
>     at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.createBeanDeserializer(BeanDeserializerFactory.java:168)
>     at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer2(DeserializerCache.java:399)
>     at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer(DeserializerCache.java:348)
>     at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:261)
>     at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:241)
>     at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
>     at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:394)
>     at com.fasterxml.jackson.databind.ObjectMapper._findRootDeserializer(ObjectMapper.java:3169)
>     at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3062)
>     at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2214)
>     at nebula.plugin.resolutionrules.ResolutionRulesPlugin.parseJsonStream(plugin.kt:147)
>     at nebula.plugin.resolutionrules.ResolutionRulesPlugin.rulesFromConfiguration(plugin.kt:94)
>     at nebula.plugin.resolutionrules.ResolutionRulesPlugin$ruleSet$2.invoke(plugin.kt:37)
>     at nebula.plugin.resolutionrules.ResolutionRulesPlugin$ruleSet$2.invoke(plugin.kt:30)
>     at kotlin.SynchronizedLazyImpl.getValue(Lazy.kt:131)
>     at nebula.plugin.resolutionrules.ResolutionRulesPlugin.getRuleSet(plugin.kt)
>     at nebula.plugin.resolutionrules.ResolutionRulesPlugin$apply$1$1.execute(plugin.kt:64)
>     at nebula.plugin.resolutionrules.ResolutionRulesPlugin$apply$1$1.execute(plugin.kt:30)
>     at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:93)
>     at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:82)
>     at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:44)
>     ... 61 more

**Comment by DanielThomas** [comment]
@PatrickAuld would you mind running the `buildEnvironment` task for me, so we have the buildscript classpath dependencies? Jackson is probably being upgraded, but the Kotlin module isn't, but just want to be sure before suggesting a solution.

**Comment by PatrickAuld** [comment]
Here you go, this is making me realize we need to do some pruning...

```
------------------------------------------------------------
Root project
------------------------------------------------------------

classpath
+--- no.nils:wsdl2java:0.5
+--- us.kirchmeier:gradle-capsule-plugin:0.10.0
+--- gradle-release:gradle-release:1.1.5
+--- org.ajoberstar:gradle-git:0.9.0
|    +--- org.ajoberstar:grgit:0.2.3
|    |    +--- org.eclipse.jgit:org.eclipse.jgit:3.3.0.201403021825-r
|    |    |    +--- com.jcraft:jsch:0.1.46 -> 0.1.49
|    |    |    +--- com.googlecode.javaewah:JavaEWAH:0.7.9
|    |    |    \--- org.apache.httpcomponents:httpclient:4.1.3 -> 4.2.5
|    |    |         +--- org.apache.httpcomponents:httpcore:4.2.4 -> 4.2.5
|    |    |         +--- commons-logging:commons-logging:1.1.1
|    |    |         \--- commons-codec:commons-codec:1.6 -> 1.8
|    |    +--- org.eclipse.jgit:org.eclipse.jgit.ui:3.3.0.201403021825-r
|    |    |    \--- org.eclipse.jgit:org.eclipse.jgit:3.3.0.201403021825-r (*)
|    |    +--- com.jcraft:jsch.agentproxy.jsch:0.0.7
|    |    |    +--- com.jcraft:jsch:0.1.49
|    |    |    \--- com.jcraft:jsch.agentproxy.core:0.0.7
|    |    +--- com.jcraft:jsch.agentproxy.pageant:0.0.7
|    |    |    +--- com.jcraft:jsch.agentproxy.core:0.0.7
|    |    |    +--- net.java.dev.jna:jna:3.4.0
|    |    |    \--- net.java.dev.jna:platform:3.4.0
|    |    +--- com.jcraft:jsch.agentproxy.sshagent:0.0.7
|    |    |    \--- com.jcraft:jsch.agentproxy.core:0.0.7
|    |    +--- com.jcraft:jsch.agentproxy.usocket-jna:0.0.7
|    |    |    +--- com.jcraft:jsch.agentproxy.core:0.0.7
|    |    |    +--- net.java.dev.jna:jna:3.4.0
|    |    |    \--- net.java.dev.jna:platform:3.4.0
|    |    +--- com.jcraft:jsch.agentproxy.usocket-nc:0.0.7
|    |    |    \--- com.jcraft:jsch.agentproxy.core:0.0.7
|    |    \--- org.slf4j:slf4j-api:1.7.6
|    \--- com.github.zafarkhaja:java-semver:0.7.2
+--- com.moowork.gradle:gradle-node-plugin:0.6
+--- org.jfrog.buildinfo:build-info-extractor-gradle:4.0.0
|    +--- commons-io:commons-io:2.0.1
|    +--- org.apache.ivy:ivy:2.2.0
|    +--- commons-logging:commons-logging:1.1.1
|    +--- commons-lang:commons-lang:2.4
|    +--- org.codehaus.groovy:groovy-all:2.3.6
|    \--- org.jfrog.buildinfo:build-info-extractor:2.5.4
|         +--- commons-io:commons-io:2.0.1
|         +--- commons-logging:commons-logging:1.1.1
|         +--- org.jfrog.buildinfo:build-info-client:2.5.4
|         |    +--- commons-io:commons-io:2.0.1
|         |    +--- org.jfrog.buildinfo:build-info-api:2.5.4
|         |    |    +--- commons-io:commons-io:2.0.1
|         |    |    +--- commons-logging:commons-logging:1.1.1
|         |    |    +--- org.codehaus.jackson:jackson-mapper-asl:1.9.12
|         |    |    |    \--- org.codehaus.jackson:jackson-core-asl:1.9.12
|         |    |    +--- com.google.guava:guava:15.0 -> 17.0
|         |    |    +--- commons-lang:commons-lang:2.4
|         |    |    \--- com.thoughtworks.xstream:xstream:1.3.1
|         |    |         \--- xpp3:xpp3_min:1.1.4c
|         |    +--- commons-codec:commons-codec:1.8
|         |    +--- commons-logging:commons-logging:1.1.1
|         |    +--- org.apache.httpcomponents:httpcore:4.2.5
|         |    +--- org.codehaus.jackson:jackson-mapper-asl:1.9.12 (*)
|         |    +--- org.apache.httpcomponents:httpclient:4.2.5 (*)
|         |    +--- com.google.guava:guava:15.0 -> 17.0
|         |    +--- commons-lang:commons-lang:2.4
|         |    \--- com.thoughtworks.xstream:xstream:1.3.1 (*)
|         +--- com.google.guava:guava:15.0 -> 17.0
|         +--- commons-lang:commons-lang:2.4
|         \--- com.thoughtworks.xstream:xstream:1.3.1 (*)
+--- com.sahlbach.gradle:gradle-jetty-eclipse-plugin:1.9.1
|    +--- org.eclipse.jetty:jetty-jsp:9.1.1.v20140108
|    |    +--- org.eclipse.jetty.toolchain:jetty-schemas:3.1.M0
|    |    +--- javax.servlet:javax.servlet-api:3.1.0
|    |    +--- javax.servlet.jsp:javax.servlet.jsp-api:2.3.1
|    |    +--- org.glassfish.web:javax.servlet.jsp:2.3.2
|    |    |    +--- org.glassfish:javax.el:[3.0.0,) -> 3.0.1-b08
|    |    |    \--- javax.servlet.jsp:javax.servlet.jsp-api:2.3.1
|    |    +--- org.eclipse.jetty.orbit:javax.servlet.jsp.jstl:1.2.0.v201105211821
|    |    |    \--- org.eclipse.jetty.orbit:javax.servlet.jsp:2.1.0.v201105211820
|    |    +--- org.eclipse.jetty.orbit:org.apache.taglibs.standard.glassfish:1.2.0.v201112081803
|    |    |    \--- org.eclipse.jetty.orbit:javax.servlet.jsp.jstl:1.2.0.v201105211821 (*)
|    |    +--- org.glassfish:javax.el:3.0.0 -> 3.0.1-b08
|    |    \--- org.eclipse.jetty.orbit:org.eclipse.jdt.core:3.8.2.v20130121
|    +--- com.google.guava:guava:15.0 -> 17.0
|    +--- org.slf4j:slf4j-api:1.6.6 -> 1.7.6
|    +--- org.eclipse.jetty:jetty-webapp:9.1.1.v20140108
|    |    +--- org.eclipse.jetty:jetty-xml:9.1.1.v20140108
|    |    |    \--- org.eclipse.jetty:jetty-util:9.1.1.v20140108
|    |    \--- org.eclipse.jetty:jetty-servlet:9.1.1.v20140108
|    |         \--- org.eclipse.jetty:jetty-security:9.1.1.v20140108
|    |              \--- org.eclipse.jetty:jetty-server:9.1.1.v20140108
|    |                   +--- javax.servlet:javax.servlet-api:3.1.0
|    |                   +--- org.eclipse.jetty:jetty-http:9.1.1.v20140108
|    |                   |    \--- org.eclipse.jetty:jetty-util:9.1.1.v20140108
|    |                   \--- org.eclipse.jetty:jetty-io:9.1.1.v20140108
|    |                        \--- org.eclipse.jetty:jetty-util:9.1.1.v20140108
|    +--- org.eclipse.jetty:jetty-plus:9.1.1.v20140108
|    |    +--- org.eclipse.jetty:jetty-webapp:9.1.1.v20140108 (*)
|    |    \--- org.eclipse.jetty:jetty-jndi:9.1.1.v20140108
|    \--- org.eclipse.jetty:jetty-annotations:9.1.1.v20140108
|         +--- org.eclipse.jetty:jetty-plus:9.1.1.v20140108 (*)
|         +--- org.eclipse.jetty:jetty-webapp:9.1.1.v20140108 (*)
|         +--- javax.annotation:javax.annotation-api:1.2
|         +--- org.ow2.asm:asm:4.1
|         \--- org.ow2.asm:asm-commons:4.1
|              \--- org.ow2.asm:asm-tree:4.1
|                   \--- org.ow2.asm:asm:4.1
+--- be.filipblondeel.gradle:gradle-gulp-plugin:0.1
|    \--- com.moowork.gradle:gradle-node-plugin:0.4 -> 0.6
+--- nl.javadude.gradle.plugins:license-gradle-plugin:0.6.0
|    +--- com.google.guava:guava:11.0.2 -> 17.0
|    \--- com.mycila.maven-license-plugin:maven-license-plugin:1.10.b1
|         +--- com.mycila.xmltool:xmltool:3.3
|         \--- org.codehaus.plexus:plexus-utils:2.0.5
+--- info.solidsoft.gradle.pitest:gradle-pitest-plugin:1.1.9
+--- org.sonarsource.scanner.gradle:sonarqube-gradle-plugin:2.0.1
|    +--- org.sonarsource.scanner.api:sonar-scanner-api:2.6
|    \--- com.google.guava:guava:17.0
\--- com.netflix.nebula:gradle-resolution-rules-plugin:1.8.0
     +--- org.jetbrains.kotlin:kotlin-stdlib:1.0.2
     |    \--- org.jetbrains.kotlin:kotlin-runtime:1.0.2
     +--- org.jetbrains.kotlin:kotlin-reflect:1.0.2
     |    \--- org.jetbrains.kotlin:kotlin-stdlib:1.0.2 (*)
     \--- com.fasterxml.jackson.module:jackson-module-kotlin:2.7.4
          +--- com.fasterxml.jackson.core:jackson-databind:2.7.4
          |    +--- com.fasterxml.jackson.core:jackson-annotations:2.7.0 -> 2.7.4
          |    \--- com.fasterxml.jackson.core:jackson-core:2.7.4
          +--- com.fasterxml.jackson.core:jackson-annotations:2.7.4
          \--- org.jetbrains.kotlin:kotlin-reflect:1.0.0 -> 1.0.2 (*)

(*) - dependencies omitted (listed previously)
```

**Comment by DanielThomas** [comment]
That looks fine (sorry we didn't get back to you). Let us know if you're still seeing this with the latest version of the plugin.

**Comment by jkschneider** [comment]
@DanielThomas We encountered this at Netflix too, as it turns out. The problem was another dependency on the buildscript classpath that was fatjaring a super old version of Jackson into its jar. The Gradle buildscript classpath nondeterministically picked up this old version of `AnnotatedConstructor`.

Drop this block into your buildscript classpath block to find the offending jar:

```groovy
buildscript.configurations.classpath.findAll {
          def file = new java.util.zip.ZipFile(it)
          file.entries().any { it.getName().contains('AnnotatedConstructor') }
}.each { println it }
```
